patches/server/0835-Call-onRemove-logic-for-breakNaturally.patch

```diff
@@ -0,0 +1,26 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Sun, 31 Oct 2021 14:49:43 -0700
+Subject: [PATCH] Call onRemove logic for breakNaturally
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
+index e3944d0f4fd18ef6c545dc4a131c0b120dff753a..2ea9a10fd18779a14dbd0421e587148ced73c734 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
+@@ -508,8 +508,14 @@ public class CraftBlock implements Block {
+             if (triggerEffect) world.levelEvent(org.bukkit.Effect.STEP_SOUND.getId(), position, net.minecraft.world.level.block.Block.getId(block.defaultBlockState())); // Paper
+             result = true;
+         }
++        // Paper start
++        net.minecraft.world.level.block.state.BlockState newBlockState = Blocks.AIR.defaultBlockState();
++        if (this.world instanceof net.minecraft.world.level.Level level) {
++            this.getNMS().onRemove(level, position, newBlockState, false);
```

Comment:
I am curious, is there any reason why you are getting the block state again here? It seems a bit higher it's stored in a variable already.

```java
  // Paper end
        // Order matters here, need to drop before setting to air so skulls can get their data
        net.minecraft.world.level.block.state.BlockState iblockdata = this.getNMS();
        net.minecraft.world.level.block.Block block = iblockdata.getBlock();
        net.minecraft.world.item.ItemStack nmsItem = CraftItemStack.asNMSCopy(item);
        boolean result = false;
```